C23_WMDE_Desktop_DE_07

Abban opened the pull request:
Var banner has a slightly different form design
and the red border now wraps everything.

Ticket: https://phabricator.wikimedia.org/T346881

tzhelyazkova commented on package-lock.json:
```diff
@@ -3925,7 +3925,8 @@
           "bundled": true,
-          "dev": true
+          "dev": true,
+          "optional": true
```
is adding the property `optional` on purpose or it just happened while running npm install?

tzhelyazkova commented:
I force pushed because there was a merge conflict in `styles.pcss`

Abban committed:
Archive C23_WMDE_Desktop_DE_06

Ticket: https://phabricator.wikimedia.org/T346881

Abban committed:
Prepare campaign for C23_WMDE_Desktop_DE_07

Ticket: https://phabricator.wikimedia.org/T346881

Abban committed:
Implement VAR for C23_WMDE_Desktop_DE_07

Var banner has a slightly different form design
and the red border now wraps everything.

Ticket: https://phabricator.wikimedia.org/T346881

Co-authored-by: Sperling-0 <tanuja.doriya@wikimedia.de>

gbirke commented on src/themes/Treedip/DonationForm/SubComponents/SelectCustomAmountRadioBubble.scss:
```diff
@@ -0,0 +1,105 @@
+	In the future, we might use tools of the CSS framework (e.g. SCSS mixins)
+	to keep the styles in sync
+	 */
+	&-select-custom-amount {
```
If you compare the hierarchy of elements in the amount selection (option elements vs custom amount element), you can see that the hierarchy is different. This class is the equivalent to the `label` element of the regular [SelectGroupRadioBubble](https://github.com/wmde/fundraising-banners/pull/228/files#diff-f5284e8cf5d1a22a5e6e669a7858c4b9e90c528685806b00dd4756114b111c82R20-R47). My approach here would be to try and apply the classes from the `label` element here (and the equivalent subclasses).
If that doesn't work correctly due to flexbox issues (in regular options the `.wmde-banner-option` is a container that has a flex attribute and determines the size of the children and that's missing here), then I'd recommend adding a div with the `wmde-banner-option` around the custom amount component.

gbirke commented on src/themes/Treedip/DonationForm/SubComponents/SelectCustomAmountRadioBubble.scss:
```diff
@@ -0,0 +1,105 @@
+		}
+
+		&-input-container {
+			border: 1px solid colors.$gray-light;
```
Remove the border here

gbirke commented on src/themes/Treedip/DonationForm/SubComponents/SelectGroupRadioBubbles.scss:
```diff
@@ -0,0 +1,67 @@
+
+.wmde-banner {
+	&-select-group {
+		&-option label {
```
To put a border around the label when its checkbox is selected, you'll need to add a class in the Vue file. For non-bubble components that class won't do anything, so it'll be backwards compatible. Preferrably add the `active` class to the `wmde-banner-select-group-option` (the element that contains the `v-for`).
The selector for that will then be underneath these definitions, something like `&-option.active label` (mind the missing dot before active, it means that the element must have *both* classes)

The reason why we have to do this in the Vue file and not in CSS is the logic of hierarchy in CSS selectors. We always say "I want to style an element that has the following parents", e.g. styling a `label` that's a child of a parent that has the `active` class: `.active label` 
Up until recently, it has been impossible to express a rule for a parent element that depends on some class or pseudoclass of a child. If we had something like a "parent" selector, let's pretend it would be "<", the you could express something like "styling a label element that has a checked checkbox" with a selector like `label < input:checked`. See this StackOverflow question form 2009: https://stackoverflow.com/questions/1014861/is-there-a-css-parent-selector
With the upcoming [:has selector](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) styling parents based on children will be possible, but that selector is quite new and not supported in Firefox. The selector would then look like `label:has(input:checked)`

moiikana commented:
rebased main
